cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -218,6 +242,9 @@ public TwillPreparer prepare(TwillApplication application) {
     labels.put(RUNNER_LABEL, RUNNER_LABEL_VAL);
     labels.put(APP_LABEL, spec.getName());
     labels.put(RUN_ID_LABEL, runId.getId());
+    if (isStatefulSetPod()) {
```

Comment:
why not be consistent and put parent pod label for any resource?

Reply:
Note - we are adding this label for any resource. The check is being done on whether the Current pod that is creating the resources is part of a stateful set - i.e AppFabric or Preview.

Reason we are not doing this for resources created by pods coming from Deployments is-  

In case of deployment resources - the names may not remain constant. 
In stateful resources - once a replica goes down, it comes up again with the same name. 

> Unlike a Deployment, a StatefulSet maintains a sticky identity for each of their Pods. 
(https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/)

Based on the above, we thought of adding this parent pod label only if the parent pod is part of stateful set.